beer-garden/functionary
# #178 - UI generic views

Closes #178 

This PR reworks what was `view_base.py` into `generic.py`, which provides extended versions for the views available in `django.views.generic` (thus the name).  There is a length module docstring that explains what the views are and how to use them here.  The short version is that they work the same as django's generic views, but with added permission checking and queryset filtering against the active environment (with some bonus nested route handling as well).

The two big changes are:
* The above mentioned `generic.py` which contains the actual view class code.
* A bunch of unit tests in `ui/tests/views/test_generic.py` that attempt to exercise the various permission checking and queryset filtering features.

I would suggest starting the review with those files.  Most everything else in this PR is just switching views over to using these new generics.

Wherever possible, the custom `test_func` methods on the view classes have been removed, since that is now implemented in the new generic views.  It is important to note that there are still cases where a custom `test_func` is required.  Similarly, there are still view classes where building off of these generics is not appropriate, so those views have not been touched.

## Test Instructions
* Unit tests as described above have been added to exercise the new generic view functionality.
* Click around the UI and make sure the pages still work.  Ideally there would have been unit tests for the existing views to give us the warm and fuzzy that these changes didn't break the existing views, but we're not there yet.
